ProcessLib/HeatTransportBHE/BHE/BHE_CXA.cpp

```diff
@@ -0,0 +1,491 @@
+          std::log(D / d_i1);
+    if (extern_Ra_Rb.use_extern_Ra_Rb)
+    {
+        _R_g = extern_Ra_Rb.ext_Rb - _R_adv_b_i1 - _R_con_i1;
+    }
+    else
+    {
+        // Eq. 69
+        _R_g = std::log(D / d_i1) / (2.0 * PI * lambda_g);
+    }
+    // Eq. 67
+    _R_con_b = chi * _R_g;
+    if (extern_Ra_Rb.use_extern_Ra_Rb)
+    {
+        _R_ff = extern_Ra_Rb.ext_Ra;
+    }
+    else if (extern_def_thermal_resistances.if_use_defined_therm_resis)
+    {
+        _R_ff = extern_def_thermal_resistances
+                    .ext_Rgg1;  // Attention! Here ext_Rgg1 is treated as Rff
+                                // for coaxial type
+    }
+    else
+    {
+        // Eq. 56
+        _R_ff = _R_adv_o1 + _R_adv_a_i1 + _R_con_o1;
+    }
+
+    // Eq. 57
+    if (extern_def_thermal_resistances.if_use_defined_therm_resis)
+        _R_fig = extern_def_thermal_resistances.ext_Rfig;
+    else
+        _R_fig = _R_adv_b_i1 + _R_con_i1 + _R_con_b;
+    // thermal resistance due to grout-soil exchange
+    if (extern_def_thermal_resistances.if_use_defined_therm_resis)
+        _R_gs = extern_def_thermal_resistances.ext_Rgs;
+    else
+        _R_gs = (1 - chi) * _R_g;
+
+    if (!std::isfinite(_R_gs))
+    {
+        OGS_FATAL(
+            "Error!!! Grout Thermal Resistance is an infinite number! The "
+            "simulation will be stopped! \n");
+    }
+}
+
+/**
+ * Nusselt number calculation
+ */
+void BHE_CXA::calcNusseltNum()
+{
+    // see Eq. 32 in Diersch_2011_CG
+
+    double Nu_in(0.0), Nu_out(0.0);
+    double gamma, xi;
+    double d_o1, d_i1, d_h;
+    double const& L = borehole_geometry.length;
+    double const& r_outer = pipe_param.r_outer;
+    double const& r_inner = pipe_param.r_inner;
+    double const& b_in = pipe_param.b_in;
+
+    d_o1 = 2.0 * r_inner;
+    d_i1 = 2.0 * r_outer;
+
+    // first calculating Nu_out
+    if (_Re_o1 < 2300.0)
+    {
+        Nu_out = 4.364;
+    }
+    else if (_Re_o1 >= 2300.0 && _Re_o1 < 10000.0)
+    {
+        gamma = (_Re_o1 - 2300) / (10000 - 2300);
+
+        Nu_out = (1.0 - gamma) * 4.364;
+        Nu_out += gamma * ((0.0308 / 8.0 * 1.0e4 * Pr) /
+                           (1.0 + 12.7 * std::sqrt(0.0308 / 8.0) *
+                                      (std::pow(Pr, 2.0 / 3.0) - 1.0)) *
+                           (1.0 + std::pow(d_o1 / L, 2.0 / 3.0)));
+    }
+    else if (_Re_o1 > 10000.0)
+    {
+        xi = pow(1.8 * std::log10(_Re_o1) - 1.5, -2.0);
+        Nu_out = (xi / 8.0 * _Re_o1 * Pr) /
+                 (1.0 + 12.7 * std::sqrt(xi / 8.0) *
+                            (std::pow(Pr, 2.0 / 3.0) - 1.0)) *
+                 (1.0 + std::pow(d_o1 / L, 2.0 / 3.0));
+    }
+
+    d_o1 = 2.0 * (r_inner + b_in);
+    d_h = 2.0 * (r_outer - (r_inner + b_in));
+    // then calculating Nu_in
+    if (_Re_i1 < 2300.0)
+    {
+        Nu_in = 3.66;
+        Nu_in += (4.0 - 0.102 / (d_o1 / d_i1 + 0.02)) * pow(d_o1 / d_i1, 0.04);
```

Comment:
Is the fraction `d_o1/d_i1` correct? Because the fractions below are inverted: `d_i1/d_o1`. Same for the Re > 10000. In CXA this is consistently `d_i1/d_o1`.

Reply:
This is not correct. We will double check it and include the CXA type of BHE in a separate PR.